### PR TITLE
feat: ability to make plots from Deep Origin dataframes

### PR DESCRIPTION
**Pull request by sg-s**
## demo

https://github.com/user-attachments/assets/c8058090-9772-4083-872c-2f9379fbecd0

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -43,7 +43,7 @@ install:
 	@python3 -m venv venv
 	@source $(CURDIR)/venv/bin/activate && \
 		pip install --upgrade pip && \
-	    pip install -e .[lint,test,jupyter,docs] && \
+	    pip install -e .[lint,test,jupyter,docs,plots] && \
```
Comment (sg-s) [inline Makefile:46]: plots are not installed by default

File: src/data_hub/plotting.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+from beartype import beartype
+from bokeh.io import show
+from bokeh.layouts import column, row
+from bokeh.models import ColumnDataSource, CustomJS, Select
+from bokeh.plotting import figure
+from deeporigin.data_hub.dataframe import DataFrame
+
+
+@beartype
+def create_interactive_scatter(df: DataFrame):
```
Comment (sg-s) [inline src/data_hub/plotting.py:12]: bokeh code is fairly inscrutable but i hope these comments help

File: src/data_hub/plotting.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    """function to make a scatter plot from a Deep Origin dataframe"""
+
+    figure_width = 500
+    select_width = int(figure_width * 0.3)
+
+    if df.shape[1] < 2:
+        raise ValueError("DataFrame must contain at least two columns.")
+
+    cols = df.attrs["metadata"]["cols"]
+    cols = [col["name"] for col in cols if col["type"] in ["float", "integer"]]
```
Comment (kennyjwilli) [inline src/data_hub/plotting.py:22]: Reuse the numeric column type list you declared elsewhere?

File: src/data_hub/plotting.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        // Update the data source
+        source.data = {'x': x_data, 'y': y_data, 'size': sizes};
+        
+        // Update the axis labels
+        x_axis.axis_label = x_select.value;
+        y_axis.axis_label = y_select.value;
+        
+        // Trigger data change
+        source.change.emit();
+    """,
```
Comment (kennyjwilli) [inline src/data_hub/plotting.py:113]: Perhaps if it gets much bigger we pull out to a js resource file and just read that file in? Then we can get some basic syntax highlighting.

**Review by kennyjwilli — APPROVED** [review]
Very cool 😎

File: src/data_hub/plotting.py (hunk 3 above)
Comment (sg-s) [reply src/data_hub/plotting.py:22]: Sadly that's it tests

File: src/data_hub/plotting.py (hunk 4 above)
Comment (sg-s) [reply src/data_hub/plotting.py:113]: Yeah. I typically put the JS in its own file as it become large.

File: src/data_hub/plotting.py (hunk 3 above)
Comment (kennyjwilli) [reply src/data_hub/plotting.py:22]: Move it? 😃

File: src/data_hub/plotting.py (hunk 2 above)
Comment (kennyjwilli) [reply src/data_hub/plotting.py:12]: They do.

**Pull request merged**